stores/editor.ts

```diff
@@ -0,0 +1,18 @@
+import { acceptHMRUpdate, defineStore } from "pinia";
+
+export interface EditorState {
+  workspace: {
+    [x: string]: any;
+  } | null;
+}
+
+export const useEditor = defineStore("editor", {
+  state: (): EditorState => ({
+    workspace: null,
```

Comment:
the semantics of "workspace" is kind of strange, workspaceSnapshot might be better